### DISCUSSION TITLE
Fix "pdm sync" installs project itself in editable mode every time

### DISCUSSION THREAD

**Discussion by linw1995** [pdm-project/pdm#562] [Ideas]
The "pdm sync" command synchronizes the packages that fit in the current env. But it keeps install the project itself in editable mode every time despite it already being installed.

And consider the scenario of project needed to build some components like C extensions, maybe we can add a new CLI flag called "---force-install-self" to indicate itself needs to be installed every time.

So without the flag, "pdm sync" will no install project itself if it is installed. (plus, with --no-editable will do install)

**Comment by frostming** [comment]
An *editable* install is not real *editable* most of time, there are quite a few scenarios that it needs re-installation:

- `console_script` change
- C extension rebuilds
- Even the dependency add/remove should update the package metadata

This is why *install self* is made the default and you can turn it off by `--no-self`, you are suggesting the opposite right? But ideally `pdm sync` (including `pdm add` and `pdm install`, `pdm remove`) shouldn't be run that often in development, it is acceptable.

**Reply by baggiponte** [reply]
I have two PDM projects, but only one of them tells me "Installing the project as an editable package" after any `pdm add`/`pdm update`... Did something change from 2.0 to 2.1, or something like that?

**Reply by frostming** [reply]
Do they both have `name` and `version` fields?

**Reply by baggiponte** [reply]
oh right, I must have messed up. now works perfectly, thank you for the prompt reply 🔝
